app/services/user_treatment.rb

```diff
@@ -0,0 +1,19 @@
+class UserTreatment
+
+  def call(user: user, type: type)
+    @type = type
+    relation = Hash.new {|h,k| h[k] = @type }
+    relation[@type.class.to_s.underscore]
+    @user  = user
+    @trim  = user.treatments.build(relation)
+
+    trim_facial_hair if type.is_a? TrimType
+    @trim
+  end
```

Comment:
I would prefer to see this method rewritten as...
```
def apply_trim(user:, trim_type:)
```
If you do add massages, then add another method...
``` def apply_massage(user:, massage_type:)```